app/connectors/httpParsers/StandardDownstreamHttpParser.scala

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package connectors.httpParsers
+
+import connectors.httpParsers.HttpParser
+import models.connectors.DownstreamOutcome
+import models.errors.{ForbiddenError, InternalError, NotFoundError}
+import models.outcomes.ResponseWrapper
+import play.api.http.Status._
+import play.api.libs.json.Reads
+import uk.gov.hmrc.http.{HttpReads, HttpResponse}
+
+object StandardDownstreamHttpParser extends HttpParser {
+
+  case class SuccessCode(status: Int) extends AnyVal
+
+  implicit def readsEmpty(implicit successCode: SuccessCode = SuccessCode(NO_CONTENT)): HttpReads[DownstreamOutcome[Unit]] =
+    (_: String, url: String, response: HttpResponse) => doRead(url, response, () => Right(ResponseWrapper(retrieveCorrelationId(response), ())))
+
+  import play.api.http.Status._
+
+  implicit def reads[A: Reads](implicit successCode: SuccessCode = SuccessCode(OK)): HttpReads[DownstreamOutcome[A]] =
+    (_: String, url: String, response: HttpResponse) =>
+      doRead(
+        url,
+        response,
+        () => {
+          val correlationId = retrieveCorrelationId(response)
+          if (response.status == OK) {
+            response.validateJson[A] match {
+              case Some(ref) => Right(ResponseWrapper(correlationId, ref))
+              case None      => Left(ResponseWrapper(correlationId, InternalError))
+            }
+          } else {
+            handleErrorResponse(correlationId, response.status)
+          }
```

Comment:
The `response.status == OK` check conflicts with the `response.status == successCode.status` in the `doRead` method which already calls `handleErrorResponse` on failure. So this can be simplified to:
```suggestion
          val correlationId = retrieveCorrelationId(response)
           response.validateJson[A] match {
            case Some(ref) => Right(ResponseWrapper(correlationId, ref))
            case None      => Left(ResponseWrapper(correlationId, InternalError))
          }
```